packages/jsii/lib/assembler.ts

```diff
@@ -501,7 +517,9 @@ export class Assembler implements Emitter {
       }
     }
 
-    const constructor = type.symbol.members && type.symbol.members.get(ts.InternalSymbolName.Constructor);
+    const constructor = (type.symbol.members && type.symbol.members.get(ts.InternalSymbolName.Constructor))
+      // If no local constructor, look for one in the erased bases
+      || (erasedBases.map(base => base.symbol.members && base.symbol.members.get(ts.InternalSymbolName.Constructor)).find(ctor => ctor != null));
```

Comment:
style: expand to multiple line - each operator in the chain in a separate line will go a long way for readability

Reply:
Actually improved this even further by factoring the "complex" expression caused by the lack of a JS "safe traversal" operator (🧐) into a separate method. This looks dope now!